attest/api/src/convert/collateral.rs

```diff
@@ -0,0 +1,90 @@
+// Copyright (c) 2023 The MobileCoin Foundation
+
+//! Convert to/from attest::Collateral
+
+use crate::{attest, ConversionError};
+use mc_attest_verifier_types::prost;
+use mc_sgx_dcap_types::Collateral;
+use mc_util_serial::Message;
+use protobuf::{CodedOutputStream, Message as ProtoMessage};
+
+impl TryFrom<&Collateral> for attest::Collateral {
+    type Error = ConversionError;
+    fn try_from(src: &Collateral) -> Result<Self, Self::Error> {
+        let prost = prost::Collateral::try_from(src)?;
+        let bytes = prost.encode_to_vec();
```

Comment:
Like the comment in `encode_to_protobuf_vec` streaming to an actual bytestream and then decoding back seemed like the best way to make sure the prost and protobuf conversion logic was the same.  I know it takes a bit more memory and time to convert, but I'm guessing actually sending over the network will greatly overshadow this conversion time